File: src/models/pagination.js
class Pagination {
  constructor() {
    this.items = null;
    this.page_range = [];
    this.has_next = false;
    this.has_prev = false;
    this.next_num = null;
    this.prev_num = null;
    this.page = null;
    this.pages = null;
    this.per_page = null;
    this.total = null;
  }
  fromData(data) {
    this.items = data.items || null;
    this.page_range = data.page_range || [];
    this.has_next = data.has_next || false;
    this.has_prev = data.has_prev || false;
    this.next_num = data.next_num || null;
    this.prev_num = data.prev_num || null;
    this.page = data.page || null;
    this.pages = data.pages || null;
    this.per_page = data.per_page || null;
    this.total = data.total || null;
  }
}

export default Pagination;

File: resources/website/api_models/pagination.py
from flask_restx import fields

from .. import swagger

pagination = swagger.model('Pagination', {
    'page': fields.Integer,
    'per_page': fields.Integer,
    'pages': fields.Integer,
    'total': fields.Integer,
    'data': fields.List(fields.Raw),
    'items': fields.Integer,
    'prev_num': fields.String,
    'next_num': fields.String,
    'has_next': fields.Boolean,
    'has_prev': fields.Boolean,
		'page_range': fields.List(fields.Raw),
})
